src/api/dataops/dataset.ts

```diff
@@ -102,6 +104,25 @@ export class Dataset<
     return new DeleteQuery<D>(this.requestExecuter, ResourceType.Dataset, this.datasetName);
   }
 
+  /**
+   * Creates an Attach query.
+   * @param  relatedDatasetName The name of the dataset to be attached.
+   * @param  filter Filtering criterion or a callback that returns one,
+   * that will be applied to the dataset to be attached.
+   * @returns AttachQuery object specialized for attaching datasets to the current one.
+   */
+  public attach(
+    relatedDatasetName: string,
```

Comment:
I would suggest to rename it to `resourceName`. It can be fileset or something else.